src/app/plugins/addclasschans.plugin.ts

```diff
@@ -139,11 +141,28 @@ export default class AddClassChannelsPlugin extends Plugin {
     this._STATE = [];
   }
 
-  private _createFirstMessage(chanName: string): MessageEmbed {
+  private async _createFirstMessage(chanName: string): Promise<MessageEmbed> {
+    const getInvChan = async () => {
+      let ret;
```

Comment:
```
    
     const  ret = this.container.guildService
        .get()
        .channels.cache.find((c) => c.name === Constants.Channels.Info.ClassInvite);
      if (!ret) {
        return (await this.container.guildService
          .get()
          .channels.fetch())
          .find((c) => c.name === Constants.Channels.Info.ClassInvite);
      }
```

Reply:
i prefer you move getInvChan to its own (private) method on the class and, keep _createFirstMessage non-async, and pass the channel id of class_invite (from getInvChan) into _createFirstMessage.

Reply:
Done. I also changed `getInvChan` to only return the channel id since that's all we use. If that's undesirable, I'll change it back.